mule-extensions-soap-api/src/main/java/org/mule/runtime/extension/api/soap/SoapServiceProvider.java

```diff
@@ -50,4 +52,15 @@
   default Map<String, String> getCustomHeaders(WebServiceDefinition definition, String operation) {
     return emptyMap();
   }
+
+  /**
+   * This method is a hook for {@link SoapServiceProvider} instances to validate the configured parameters and
+   * fail gracefully before attempting to create a connection avoiding misleading and confusing error messages.
+   *
+   * @return a {@link ConnectionValidationResult#failure} if the configuration is invalid,
+   * {@link ConnectionValidationResult#success()} otherwise.
+   */
+  default ConnectionValidationResult validate() {
```

Comment:
If this is just validating config parameters then it shouldn't return a ConnectionValidationResult. that type implies that connectivity testing was performed. Which was not the case here.